src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,87 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int MULTIPLIER = 2;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold) {
+            resize();
+        }
+        int index = getIndex(key);
+        Node<K, V> newNode = table[index];
+        if (newNode == null) {
+            table[index] = new Node<>(key, value, null);
+        } else {
+            while (newNode.next != null
+                    || Objects.equals(key, newNode.key)) {
+                if (Objects.equals(key, newNode.key)) {
```

Comment:
Can we simplify these ?
|| Objects.equals(key, newNode.key)) {
                if (Objects.equals(key, newNode.key))